code/modules/food_and_drinks/machinery/coffeemaker.dm

```diff
@@ -140,9 +145,17 @@
 		cartridge = null
 	update_appearance()
 
-/obj/machinery/coffeemaker/update_icon_state()
-	icon_state = "[base_icon_state][coffeepot ? "_pot" : "_nopot"][cartridge ? "_cart": "_nocart"]"
-	return ..()
+/obj/machinery/coffeemaker/update_overlays()
+	. = ..()
+	. += overlay_checks()
+
+/obj/machinery/coffeemaker/proc/overlay_checks()
+	. = list()
+	if(coffeepot)
+		. += "coffeemaker_pot"
+	if(cartridge)
+		. += "coffeemaker_cartidge"
+	return
```

Comment:
```suggestion
	return .
```

New procs should aim to make `.` usage explicit, but you can also just make this `var/list/extra_overlays` since the `.` provides no safety or clarity here.